### PR TITLE
Add unit test for SQLTokenGenerators (#14912)

### LINKED ISSUE #14912: Add more unit test for SQLTokenGenerators
Hi, community:

This issue is to add more unit tests for the SQLTokenGenerators class. Welcome to claim them.

The specific subtasks are as follows:

- [ ] Add unit test for SQLTokenGenerators#addAll
- [ ] Add unit test for SQLTokenGenerators#generateSQLTokens

### PR DESCRIPTION
**Pull request by jiangtao69039**
* Add unit test for SQLTokenGenerators#addAll

* Add unit test for SQLTokenGenerators#generateSQLTokens

Fixes #14912 .

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-rewrite/src/test/java/org/apache/shardingsphere/infra/rewrite/sql/token/generator/SQLTokenGeneratorsTest.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        sqlTokenGenerators.addAll(Collections.singleton(mockSqlTokenGenerator));
+        assertThat(actualSqlTokenGeneratorsMap.size(), is(1));
+        assertTrue(actualSqlTokenGeneratorsMap.containsKey(mockSqlTokenGenerator.getClass()));
+        assertThat(actualSqlTokenGeneratorsMap.get(mockSqlTokenGenerator.getClass()), is(mockSqlTokenGenerator));
+    }
+
+    @Test
+    public void assertAddAllWithSameClass() throws Exception {
+        SQLTokenGenerators sqlTokenGenerators = new SQLTokenGenerators();
+        SQLTokenGenerator firstItem = mock(SQLTokenGenerator.class);
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-rewrite/src/test/java/org/apache/shardingsphere/infra/rewrite/sql/token/generator/SQLTokenGeneratorsTest.java:62]: Can we use more meaningful param name?

File: shardingsphere-infra/shardingsphere-infra-rewrite/src/test/java/org/apache/shardingsphere/infra/rewrite/sql/token/generator/SQLTokenGeneratorsTest.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void assertGenerateSQLTokensWithOptionalSQLTokenGenerator() {
+        OptionalSQLTokenGenerator<SQLStatementContext> optionalSQLTokenGenerator = mock(OptionalSQLTokenGenerator.class);
+        SQLTokenGenerators sqlTokenGenerators = new SQLTokenGenerators();
+        sqlTokenGenerators.addAll(Collections.singleton(optionalSQLTokenGenerator));
+        SQLToken expectedToken = mock(SQLToken.class);
+        when(optionalSQLTokenGenerator.generateSQLToken(any(SQLStatementContext.class))).thenReturn(expectedToken);
+        List<SQLToken> actualSqlTokens = sqlTokenGenerators.generateSQLTokens(mock(SQLStatementContext.class), new ArrayList<>(), mock(ShardingSphereSchema.class));
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-rewrite/src/test/java/org/apache/shardingsphere/infra/rewrite/sql/token/generator/SQLTokenGeneratorsTest.java:91]: Collection may be better.

**Review by strongduanmu — CHANGES_REQUESTED** [review]
@rfscholte Thank you for your contribution, this pr looks good, only have some code style that needs to be optimized.

File: shardingsphere-infra/shardingsphere-infra-rewrite/src/test/java/org/apache/shardingsphere/infra/rewrite/sql/token/generator/SQLTokenGeneratorsTest.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SQLTokenGeneratorsTest {
```
[the remark this answers is not in the document]
Comment (jiangtao69039) [reply shardingsphere-infra/shardingsphere-infra-rewrite/src/test/java/org/apache/shardingsphere/infra/rewrite/sql/token/generator/SQLTokenGeneratorsTest.java:46]: got it

**Review by strongduanmu — APPROVED** [review]
@jiangtao69039 Good job, thank you.

**Pull request merged**